Examples/Algorithms/TrackFitting/src/GsfFitterFunction.cpp

```diff
@@ -106,23 +111,38 @@ std::shared_ptr<TrackFittingAlgorithm::TrackFitterFunction>
 ActsExamples::makeGsfFitterFunction(
     std::shared_ptr<const Acts::TrackingGeometry> trackingGeometry,
     std::shared_ptr<const Acts::MagneticFieldProvider> magneticField,
+    std::string lowBetheHeitlerPath, std::string highBetheHeitlerPath,
     std::size_t maxComponents, bool abortOnError,
     bool disableAllMaterialHandling) {
   MultiStepper stepper(std::move(magneticField));
 
+  const auto bhapp = [&]() {
+    if (lowBetheHeitlerPath.empty() && highBetheHeitlerPath.empty()) {
+      return Acts::Experimental::makeDefaultBetheHeitlerApprox();
+    } else if (std::filesystem::exists(lowBetheHeitlerPath) &&
+               std::filesystem::exists(highBetheHeitlerPath)) {
+      return Acts::Experimental::AtlasBetheHeitlerApprox<6, 5>::loadFromFile(
+          lowBetheHeitlerPath, highBetheHeitlerPath);
+    } else {
+      throw std::invalid_argument(
+          "Paths to bethe heitler parameterization do not exist. Pass empty "
+          "strings to load a default parameterization");
+    }
```

Comment:
wouldn't it be enough to just have the return statement as this would bubble up an exception anyway? if you want to be more descriptive of the error you can catch and throw again

Reply:
Hmm yeah I wanted to document the behavior in this situation. However, this is not optimal, because it is somehow counter-intuitive that empty string is valid and wrong path not... Maybe std::optional<std::string>? But how would this be on the python side? I could also expose the Bethe heitler object to python... that would be the clearest solution I guess... What is your opinion on this @andiwand ?